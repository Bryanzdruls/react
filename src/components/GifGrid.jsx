import React, { useEffect } from 'react'
import { getGifs } from '../helpers/GetGifs';

export const GifGrid = ({category}) => {
 
  useEffect(()=>{
    getGifs( category );
  },[])


  return (
    <>
        <h3> {category} </h3>        
    </>
  )
}
