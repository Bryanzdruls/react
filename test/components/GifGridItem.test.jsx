
import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Probando GifGrid TArea', () => {
  const title ='naruto';
  const url ='https://www.luffy.com/onepiece';
  
  
  test('Deberia recibir titulo y url', () => {
    render(<GifGridItem title={title} url={url}></GifGridItem>);
    expect(render).toMatchSnapshot();
  })

  test('debe de mostrar una imagen coin el URL y el ALT indicado ', () => {
    render(<GifGridItem title={title} url={url}></GifGridItem>);
    
    screen.debug();
    /* expect( screen.getByRole('img').src ).toBe( url );
    expect( screen.getByRole('img').alt ).toBe( title ); */
    const {src, alt }=screen.getByRole( 'img' );
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  
  })
  
  test('Debe de mostrar que el titulo existe', () => {
    render( <GifGridItem title={ title } url={ url } /> );
    expect( screen.getByText( title ) ).toBeTruthy();
    
  })
  
})
