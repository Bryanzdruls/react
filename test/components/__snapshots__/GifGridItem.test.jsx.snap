// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGrid TArea Deberia recibir titulo y url 1`] = `[Function]`;
